compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -436,6 +436,17 @@ PadEncodingLayoutAttr PadEncodingLayoutAttr::getIdentityAttr(MLIRContext *ctx,
   return get(ctx, zeros);
 }
 
+Attribute
+PadEncodingLayoutAttr::cloneWithLayouts(ArrayRef<Attribute> layouts) const {
+  MLIRContext *ctx = getContext();
+  return LayoutAttr::get(ctx, ArrayAttr::get(ctx, layouts));
+}
+
+bool PadEncodingLayoutAttr::isSerialized() const {
+  return llvm::all_of(getPadding().asArrayRef(),
+                      [](int64_t val) { return val != ShapedType::kDynamic; });
```

Comment:
nit: this can be `return !ShapedType::isDynamicShape(getPadding().asArrayRef())`